### PR TITLE
Avoid deleting scale set if annotation is not parsable or if it does not exist

### PR DESCRIPTION
**Pull request by nikola-jokic**
If autoscaling runner set is applied without secret, the controller will not be able to create scale set.
It does not annotate the resource with an ID.

If you try to manually delete the autoscaling runner set, the controller will get stuck reporting an error trying to parse scale set id.
This change skips the deletion if annotation contains invalid ID or if it does not exist. In that case, deletion of a resource will proceed and cleanup would be successful

### CODE REVIEW

File: controllers/actions.github.com/autoscalingrunnerset_controller.go [hunk 1]
```diff
@@ -412,8 +412,11 @@ func (r *AutoscalingRunnerSetReconciler) deleteRunnerScaleSet(ctx context.Contex
 	logger.Info("Deleting the runner scale set from Actions service")
 	runnerScaleSetId, err := strconv.Atoi(autoscalingRunnerSet.Annotations[runnerScaleSetIdKey])
 	if err != nil {
-		logger.Error(err, "Failed to parse runner scale set ID")
-		return err
+		// If the annotation is not set correctly, or if it does not exist, we are going to get stuck in a loop trying to parse the scale set id.
+		// If the configuration is invalid (secret does not exist for example), we never get to the point to create runner set. But then, manual cleanup
+		// would get stuck finalizing the resource trying to parse annotation indefinitely
+		logger.Info("autoscaling runner set does not have annotation describing scale set id. Skip deletion")
```
Comment (TingluoHuang) [inline controllers/actions.github.com/autoscalingrunnerset_controller.go]: do we want to add an L0 for this case?
should we trace the `err` out?

File: controllers/actions.github.com/autoscalingrunnerset_controller.go [hunk 2]
```diff
@@ -412,9 +412,13 @@ func (r *AutoscalingRunnerSetReconciler) deleteRunnerScaleSet(ctx context.Contex
...
 	if err != nil {
-		logger.Error(err, "Failed to parse runner scale set ID")
-		return err
+		// If the annotation is not set correctly, or if it does not exist, we are going to get stuck in a loop trying to parse the scale set id.
+		// If the configuration is invalid (secret does not exist for example), we never get to the point to create runner set. But then, manual cleanup
+		// would get stuck finalizing the resource trying to parse annotation indefinitely
+		logger.Info("autoscaling runner set does not have annotation describing scale set id. Skip deletion", "err", err.Error())
+		return nil
 	}
+	fmt.Println("Well, that is wierd")
```
Comment (rentziass) [inline controllers/actions.github.com/autoscalingrunnerset_controller.go]: A debug survivor? :)

Comment (nikola-jokic) [reply controllers/actions.github.com/autoscalingrunnerset_controller.go]: hahaha yeah, sorry for that :smile:

Comment (rentziass) [inline controllers/actions.github.com/autoscalingrunnerset_controller.go]:
```suggestion
```

**Review by rentziass — DISMISSED** [review]

**Review by rentziass — APPROVED** [review]

**Pull request merged**